docs/data-analysis.md

```diff
@@ -1,164 +1,92 @@
-data.tail()
-
-```
-
-### Explanation
-
-#### Imports and constant definition
-
-``` python
-# Some necessary imports
-from pathlib import Path
+from freqtrade.data.btanalysis import load_backtest_data
+from freqtrade.data.btanalysis import load_trades_from_db
 
-from freqtrade.data.history import load_pair_history
-from freqtrade.resolvers import StrategyResolver
 # Define some constants
-ticker_interval = "5m"
-
+ticker_interval = "1m"
 # Name of the strategy class
-strategyname = 'Awesomestrategy'
+strategy_name = 'NewStrategy'
+# Path to user data
+user_data_dir = 'user_data'
 # Location of the strategy
-strategy_location = 'user_data/strategies'
+strategy_location = os.path.join(user_data_dir, 'strategies')
 # Location of the data
-data_location = 'user_data/data/binance'
+data_location = os.path.join(user_data_dir, 'data', 'binance')
+# Pair to analyze 
 # Only use one pair here
-pair = "XRP_ETH"
+pair = "BTC_USDT"
 ```
 
-This first section imports necessary modules, and defines some constants you'll probably need to adjust for your case.
+### Load exchange data
 
-#### Load candles
-
-``` python
-# Load data
+```python
+# Load data using values set above
 bt_data = load_pair_history(datadir=Path(data_location),
-                            ticker_interval = ticker_interval,
+                            ticker_interval=ticker_interval,
                             pair=pair)
-print(len(bt_data))
+
+# Confirm success
+print("Loaded " + str(len(bt_data)) + f" rows of data for {pair} from {data_location}")
 ```
 
-This second section loads the historic data and prints the amount of candles in the DataFrame.
-You can also inspect this dataframe by using `bt_data.head()` or `bt_data.tail()`.
+### Load and run strategy  
 
-#### Run strategy and analyze results
+* Rerun each time the strategy file is changed
+* Display the trade details. Note that using `data.head()` would also work, however most indicators have some "startup" data at the top of the dataframe.
 
-Now, it's time to load and run your strategy.
-For this, I recommend using a new cell in your notebook, since you'll want to repeat this until you're satisfied with your strategy.
+Some possible problems:
 
-``` python
-# Load strategy - best done in a new cell
-# Needs to be ran each time the strategy-file is changed.
-strategy = StrategyResolver({'strategy': strategyname,
-                            'user_data_dir': Path.cwd(),
-                            'strategy_path': location}).strategy
+* Columns with NaN values at the end of the dataframe
+* Columns used in `crossed*()` functions with completely different units
+
+```python
+# Load strategy using values set above
+strategy = StrategyResolver({'strategy': strategy_name,
+                            'user_data_dir': user_data_dir,
+                            'strategy_path': strategy_location}).strategy
 
 # Run strategy (just like in backtesting)
 df = strategy.analyze_ticker(bt_data, {'pair': pair})
```

Comment:
`analyze_ticker` does not export to `user_data/backtest_data/backtest-result.json` like the `Backtesting` class. Therefore, `df = load_backtest_data("user_data/backtest_data/backtest-result.json")` either fails or returns stale results.

What would you suggest?

Reply:
well, `analyze_ticker()` is not a replacement for backtesting - the only thing it does is give you a dataframe with your indicators and buy/sell signals, so you can plot / analyze/ ....

I guess the comment "just like in backtesting" is misleading and maybe should be replaced with "populate indicators and buy/sell signals" - letme explain:

**example**
having 200 buy signals as output for one pair from `analyze_ticker()` (`assert df['buy'].sum() == 200`)  does not mean that 200 trades will be made during backtesting (unless --eps and --dmmp - the "unrealistic" flags) are used.

**Explanation:**
Assuming you use only one condition - `df['rsi'] < 30` as buy condition - this will generate multiple "buy" signals for each pair in sequence (until rsi returns > 29).
The bot will only buy on the first of these signals (and also only if a trade-slot ("max_open_trades") is still available), or on one of the middle signals, as soon as a "slot" becomes available.

We currently can't call backtesting from notebooks for a few reasons (it'll crash ...) - but i guess it's just a small step to enable that in the future.

Reply:
This is ugly but works in a notebook:

```
import os
import freqtrade
import subprocess

# Configure cli command as a python list of strings
cmd = ['freqtrade', 
       '-c', Path(str(Path.home()),'.freqtrade/exchange-config.json'), 
       '-c', 'user_data/user_repo/config.json', 
       'backtesting', "--refresh-pairs-cached",
      "--timerange=-200",
      "--export", "trades",
      "--export-filename=user_data/backtest_data/backtest-result.json"
     ]

# Run command using subprocess
try:
    # If command succeeds capture output
    normal = subprocess.run(cmd,
        stdout=subprocess.PIPE, stderr=subprocess.PIPE,
        check=True,
        text=True)
except:
    # If command fails capture shell error instead of python error
    normal = subprocess.run(cmd,
        stdout=subprocess.PIPE, stderr=subprocess.PIPE,
        text=True)
    
# Print captured output
print(normal.stdout)
print(normal.stderr)
```

Reply:
but it defeats the purpose of using a jupyter notebook in my eyes.
so while I'm sure it works, i'm not sure it's useful to have it documented that way, since it'll be quite confusing to users.

It's better to wait a bit and have the proper way working and documented (as import)

Reply:
> It's better to wait a bit and have the proper way working and documented (as import)

I agree that introducing this to users would be confusing. Take a look at the example notebook to see how I proposed clarifying this issue until backtests can be run in a notebook.

Reply:
Yep, that description looks good :+1: